uspv/init.go

```diff
@@ -13,67 +13,48 @@ import (
 	"github.com/mit-dci/lit/lnutil"
 )
 
-// Connect dials out and connects to full nodes.
-func (s *SPVCon) Connect(remoteNode string) error {
+func (s *SPVCon) GetSeedAdrs(seed string) ([]string, error) {
 	var err error
+	var seedAdrs []string // slice of IP addrs returned from the DNS seed
+	log.Printf("Attempting to retrieve peers to connect to based on DNS Seed\n")
 
-	// slice of IP addrs returned from the DNS seed
-	var seedAdrs []string
-	// if remoteNode is "yes" but no IP specified, use DNS seed
-	if lnutil.YupString(remoteNode) {
-		// connect to DNS seed based node
-		log.Printf("Attempting connection based on DNS seed\n")
-		if len(s.Param.DNSSeeds) == 0 {
-			// no available DNS seeds
-			return fmt.Errorf(
-				"Can't connect: No known DNS seeds for %s", s.Param.Name)
-		}
-
-		// go through seeds to get a list of IP addrs
-		for _, seed := range s.Param.DNSSeeds {
-			seedAdrs, err = net.LookupHost(seed)
-			if err == nil {
-				// got em
-				log.Printf("Got %d IPs from DNS seed %s\n", len(seedAdrs), seed)
-				break
-			}
-			// gotta keep trying for those IPs
-			log.Printf("DNS seed %s error", seed)
-		}
+	seedAdrs, err = net.LookupHost(seed)
+	if err == nil {
+		log.Printf("Got %d IPs from DNS seed %s\n", len(seedAdrs), seed)
+		// got seedAdrs, but it may be empty
+	} else {
+		return nil, fmt.Errorf("Have difficulty trying to conenct to %s. Going to the next seed", seed)
+	}
 
-		if len(seedAdrs) == 0 {
-			// never got any IPs from DNS seeds; give up
-			return fmt.Errorf(
-				"Can't connect: No functioning DNS seeds for %s", s.Param.Name)
-		}
-		// now have some IPs, go through and try to connect to one.
-		var connected bool
-		for _, ip := range seedAdrs {
-			log.Printf("Attempting connection to node at %s\n",
-				ip+":"+s.Param.DefaultPort)
-			s.con, err = net.Dial("tcp", ip+":"+s.Param.DefaultPort)
-			if err != nil {
-				log.Println(err.Error())
-			} else {
-				connected = true
-				break
-			}
-		}
-		if !connected {
-			return fmt.Errorf("Tried all IPs from DNS seed, none worked")
-		}
-	} else { // else connect to user-specified node
-		if !strings.Contains(remoteNode, ":") {
-			remoteNode = remoteNode + ":" + s.Param.DefaultPort
-		}
+	if len(seedAdrs) == 0 {
+		// check whether seedAdrs is not empty
+		return nil, fmt.Errorf("No peers found conencted to %s. Continuing.", seed)
+	}
+	return seedAdrs, nil
+	// return the seedAdrs here
+}
 
-		// open TCP connection to specified host
-		s.con, err = net.Dial("tcp", remoteNode)
-		if err != nil {
-			return err
+func (s *SPVCon) DialNode(seedAdrs []string) error {
```

Comment:
Doesn't (or shouldn't) need the argument here

Reply:
Nevermind, this is needed -- instead maybe change the name, as it's not a seedAdrs slice, since it might not be from a seed.  Just nodeAdrs?

Reply:
yeah cool